specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,522 @@
+| __quantum__rt__tuple_record_output  | `void(i64,i8*)`      | Inserts a marker in the generated output that indicates the start of a tuple and how many tuple elements it has. The second parameter defines a string label for the tuple. Depending on the output schema, the label is included in the output or omitted.  |
+| __quantum__rt__array_record_output  | `void(i64,i8*)`      | Inserts a marker in the generated output that indicates the start of an array and how many array elements it has. The second parameter defines a string label for the array. Depending on the output schema, the label is included in the output or omitted. |
+| __quantum__rt__result_record_output | `void(%Result*,i8*)` | Adds a measurement result to the generated output. The second parameter defines a string label for the result value. Depending on the output schema, the label is included in the output or omitted.                                                         |
+
+### Initialization
+
+*A [workstream](https://github.com/qir-alliance/qir-spec/issues/11) to specify
+how to initialize the execution environment is currently in progress. As part of
+that workstream, this paragraph and the listed initialization function(s) will
+be updated.*
+
+### Output Recording
+
+The program output of a quantum application is defined by a sequence of calls to
+runtime functions that record the values produced by the computation,
+specifically calls to the runtime functions ending in `record_output` listed in
+the table [above](#runtime-functions). In the case of the Base Profile, these
+calls are contained within the final block of the entry point function, i.e. the
+block that terminates in a return instruction.
+
+For all output recording functions, the `i8*` argument must be a non-null
+pointer to a global constant that contains a null-terminated string. A backend
+may ignore that argument if it guarantees that the order of the recorded output
+matches the order defined by the entry point. Conversely, certain output schemas
+do not require the recorded output to be listed in a particular order. For those
+schemas, the `i8*` argument serves as a label that permits the compiler or tool
+that generated the labels to reconstruct the order intended by the program.
+[Compiler frontends](https://en.wikipedia.org/wiki/Compiler#Front_end) must
+always generate these labels in such a way that the bitcode does not depend on
+the output schema; while choosing how to best label the program output is up to
+the frontend, the choice of output schema on the other hand is up to the
+backend. A backend may reject a program as invalid or fail execution if a label
+is missing.
+
+Both the labeling schema and the output schema are identified by a metadata
```

Comment:
I only see the `output_labeling_schema` attribute defined, but this comment makes it seem like there should be another attribute for the output schema. Am I misinterpreting that? (Note that lines 338-339 say that `output_labeling_schema` is for the labeling schema (and hence not the output schema, presumably).